src/Tests/Microsoft.NET.Sdk.Razor.Tests/AspNetSdkBaselineTest.cs

```diff
@@ -356,8 +386,36 @@ string UpdateAssetVersion(string restorePath, string property)
                     .Should()
                     .BeEquivalentTo(expected.ReferencedProjectsConfiguration.OrderBy(cm => cm.Identity));
                 manifest.DiscoveryPatterns.OrderBy(dp => dp.Name).ShouldBeEquivalentTo(expected.DiscoveryPatterns.OrderBy(dp => dp.Name));
-                manifest.Assets.OrderBy(a => a.BasePath).ThenBy(a => a.RelativePath).ThenBy(a => a.AssetKind)
-                    .ShouldBeEquivalentTo(expected.Assets.OrderBy(a => a.BasePath).ThenBy(a => a.RelativePath).ThenBy(a => a.AssetKind));
+
+                var manifestAssets = manifest.Assets.OrderBy(a => a.BasePath).ThenBy(a => a.RelativePath).ThenBy(a => a.AssetKind);
+                var expectedAssets = expected.Assets.OrderBy(a => a.BasePath).ThenBy(a => a.RelativePath).ThenBy(a => a.AssetKind);
+                
+                manifestAssets.ShouldBeEquivalentTo(expectedAssets, AssetDifferencesDetails(manifestAssets, expectedAssets));
+
+                static string AssetDifferencesDetails(IEnumerable<StaticWebAsset> manifestAssets, IEnumerable<StaticWebAsset> expectedAssets)
```

Comment:
How does this compare to the existing reported error message? Could you provide an example of the before and after?

I believe that when we compare the collections the default message is pretty good already and tells you about this information in more detail.

Reply:
> I believe that when we compare the collections the default message is pretty good already and tells you about this information in more detail.

That isn't what I've seen. Below is a typical output (eg from [here](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-sdk-refs-pull-27149-merge-b7e367a1ef28449ab4/Microsoft.NET.Sdk.BlazorWebAssembly.Tests.dll.1/3/console.3a7305ce.log?helixlogtype=result))

```
    Microsoft.NET.Sdk.BlazorWebAssembly.Tests.BlazorWasmStaticWebAssetsIntegrationTest.StaticWebAssets_PublishMinimal_Works [FAIL]
      Expected subject to be a collection with 239 item(s), but found 236.
```
Sometimes it dumps the two collections in a blob of JSON. That's hard enough to read - you have to parse it - but the biggest problem is that it only shows the first N elements of each, and ends with "...". The couple of times I have taken these JSON blobs, broken them into lines, and diffed, they have been identical : the differences were hidden by the "...".

So in the cases I've seen, it's necessary to pull down the change, build the repo and run the test yourself.

Reply:
@danmoseley I think I see what you mean.

I played a bit with it, and the comparer first checks the collection size and then checks item differences that it finds.

That said, I think we are better of changing this to avoid using `ShouldBeEquivalentTo` altogether. It was done for convenience here I think, but the results are not ideal. 

The problem with the current change is that it does not format things correctly when a property in an item changes (which was why we were using `ShouldBeEquivalentTo`.

I think it would work better if we did the collection comparison ourselves "by hand" and produce the appropriate messages instead. The key aspect is whether we want to display all the differences in the collection at once or only the first difference.

What I want to make sure is that we don't lose the ability to see changes to individual items.

Reply:
I see what you're saying here how property specific diffs being printed would be more helpful. Till we get a chance to do that though, would it be fine to merge this as is? I feel it still provides more context than we have presently on the specific files that are new/missing.

Reply:
I do think that this helps in a specific case but breaks other cases. Would it be so much work to replace the comparison to do it manually?

Reply:
Is this what 86747ab4842eb986e31bc88f8048d6419560b8c6 you had in mind by a property level comparison?